# M01_E03_Intro_Python_Bit_ASCII_Formatear
Ascii_TextFormating

### PENSAMIENTO COMPUTACIONAL 2024-B

* Prof. Ricardo Vega M. - Escuela de Diseño Universidad Católica de Chile - 2024.  

* Renombrar archivo así: E02_apellido_nombre_rut, por ejemplo:
E0X_vega_ricardo_102349285  

* Renombrar el titulo así: M01_E0X_PRINT_SUMAR_apellido_nombre_rut, por ejemplo:
M0X_E0X_Print_Sumar_vega_ricardo_102349285

* X equivale al número de la tarea.

### En la tarea de esta semana, veremos estos temas:

* 1- Bits!  
* 2- Codificación ASCII.  
* 3- Formatear Texto.  


## A- Números Binarios! 

**Sobre los bits y números binarios**  
Ya sabemos que el computador trabaja con bits, la unidad de datos mas básica. Y ocho bits forman un byte. Toma valores 0 o 1, y se cuentan de derecha a izquierda, como los números decimales. Tiene un valor posicional que va aumentando x2: 128, 64, 32, 16, 8, 4, 2, 1. Por ejemplo la notación binaria 00100101 = 37 en notación decimal. 

En python para escribir binario se escribe "0b" y luego el octeto, así: 0b00100101   


In [39]:
# Al hacer print aparece el valor
print(0b00100101)

37


## 1- Fecha nacimiento en binario (1 punto)  

**Descripción**  

En esta tarea deberá escribir en números binarios su fecha de nacimiento, cada dígito separado. Para los años, como los octetos pueden representar hasta 1024 solamente, usaremos 2 variables para sumarlas y llegar a los años sobre 1024. Y finalmente, concatenarlos dentro de un print(). Recuerde que concatenar es poner valores uno al lado de otro. 

Recuerde: En python para escribir binario se escribe "0b" y luego el octeto, así: 0b00100101   


In [40]:
## Ejemplo 1- Fecha nacimiento en binario. 
## Fecha nacimiento, cada fecha en variable separada
## Debe usar dos para el año, pues los octetos llegan solo hasta 1024.
## Recuerde que puede copiar este ejemplo y cambiar los valores. 

d = 0b00100001
m = 0b01101
a1 = 0b1111101000
a2 = 0b1111001001

print( d, m, a1 + a2 ) 


33 13 1969


In [10]:
## Su respuesta desde aca.

## B- Codificación ASCII.  

**Sobre la funciones ord() y chr()**  

En el computador los caracteres alfanuméricos (números, letras, símbolos, etc.) son  valores numéricos binarios, que el computador, mediante un proceso de codificación y decodificación, muestra como los símbolos que vemos por pantalla. Por ejemplo:  

In [25]:
## Run para mostrar ejemplo. 
print( chr(97) )  # funcion ascii() muestra valor alfanumérico del caracter: a  
print( ord("a") ) # funcion ord() muestra valor ascii del caracter: 97


a
97


### 2– Caracter a ASCII (1 punto) 

**Descripción**  

En esta tarea deberá codificar su nombre o apellido, pasando de carácter a ascii con la función ord(). Cada letra de la palabra en línea separada, y luego concatenar en un print(). 

In [28]:
## Ejemplo 2– Caracter a ASCII (1 punto) 
## Codificar de caracter alfanumérico a ascii, usando bajas o altas, esto es libre. 
## Use cuantas variables necesite según las letras de su nombre o apellido. 
## Recuerde que puede copiar este ejemplo y cambiar los valores. 

a = ord("v")
b = ord("e")
c = ord("g")
d = ord("a")

# mostrará los valores en ascii de cada carácter. 
print( a, b, c, c ) # en este caso usamos comas para separar elementos, no usamos +. 


118 101 103 103


In [ ]:
## Su respuesta desde acá.  

### 3– ASCII a Caracter (1 punto) 

**Descripción**  
Codificar una de las palabras de abajo a carácter alfanumérico usando la función chr(). Cada código ascii de la palabra en línea separada, y luego concatenar en un print(). La palabra que le corresponde es según el último número de su rut (todas son de 8 letras).     
* 0 =  97, 109, 98, 105, 101, 110, 116, 101
* 1 = 100, 101, 114, 101, 99, 104, 111, 115
* 2 = 101, 116, 105, 113, 117, 101, 116, 97
* 3 = 103, 111, 98, 105, 101, 114, 110, 111
* 4 = 104, 105, 115, 116, 111, 114, 105, 97   
* 5 = 109, 105, 108, 108, 111, 110, 101, 115   
* 6 = 110, 97, 99, 105, 111, 110, 97, 108   
* 7 = 112, 114, 111, 103, 114, 97, 109, 97   
* 8 = 114, 101, 99, 117, 101, 114, 100, 97   
* 9 = 115, 105, 108, 101, 110, 99, 105, 111   
* K = 116, 114, 97, 98, 97, 106, 97, 114   


In [30]:
## Ejemplo 3– ASCII a Caracter. 
## Codificar de ascii, a alfanumérico usando chr().  
## Recuerde que puede copiar este ejemplo y cambiar los valores. 

x1 = chr(112)
x2 = chr(101)
x3 = chr(114)
x4 = chr(115)
x5 = chr(111)
x6 = chr(110)
x7 = chr(97)
x8 = chr(115)

# mostrará valores alfanuméricos. 
print( x1, x2, x3, x4, x5, x6, x7, x8 ) # en este caso usamos comas para separar elementos, no usamos +. 


p e r s o n a s


In [ ]:
## Su respuesta desde acá.

## C- Formatear Texto en Consola.  

Cuando usamos la función print() mostramos texto en la consola, que es el espacio debajo de cada celdita, donde aparece la información de los valores que queremos mostrar después de alguna operación. 

Al colocar el texto (string) entre triple comillas, este queda con el formato y espacios que se usan dentro de las comillas. Pueden usar comillas simples o dobles, pero las mismas en el inicio y en el cierre. 


### 4- Concatenar Frase (1 punto)

**Descripcion**  

Construir la frase de manera que se vea el resultado esperado, e imprimirlo en consola con print(). Debe leerse "Al sumar K y N el resultado es Z" 


In [32]:
## Ejemplo 4- Concatenar Frase  
## Qué va en cada ? para que resulte "Al sumar K y N el resultado es Z"
## Recuerde que puede copiar este ejemplo y cambiar los valores. 

K = ? # acá su edad
N = ? # acá su año de nacimiento
Z = ? + ? 

txt1 = "Al sumar"
suma = ?
txt3 = "el resultado es"

print( ?, ?, ?, ?, ?, ? ) 


SyntaxError: invalid syntax (2700911585.py, line 5)

In [ ]:
## Su respuesta desde acá.

### 5- Texto Multilínea (1 punto). 

**Descripción** 

Escriba su nombre completo ( todos sus nombres y apellidos) formateado en varias líneas de manera escalonada, e imprimirlo en consola con print() 


In [37]:
## Ejemplo 3B- Texto Multilíinea. 
## Se puede formatear texto usando tres comillas
## las mismas al inicio y al final.  
## Recuerde que puede copiar este ejemplo y cambiar los valores. 

lineas = """Este
    es un 
        texto
            fome
 ...
     y     malo"""

print( lineas ) 


Este
    es un 
        texto
            fome
 ...
     y     malo


In [ ]:
## Su respuesta desde acá.

### 6– Dividir frase poema (1 punto). 

Deberá escoger aleatoriamente una línea del poema "La Araucana" (A. de Ercilla), y dividir cada línea en las diferentes palabras que la componen. Usar el caracter especial \n en donde las letras se separan. Luego imprimirlo en consola con print(). 

Poema de en este sitio: https://es.wikisource.org/wiki/La_araucana_primera_parte/I 

El carácter \n es invisible, separa caracteres en una nueva linea (newline). 

In [41]:
## Ejemplo 6– Dividir frase poema.  
## El carácter especial \n se usa para salto de línea, es invisible
## No aparece en la pantalla.
## Recuerde que puede copiar este ejemplo y cambiar los valores. 

araucana_linea = "No \nlas \ndamas, \namor, \nno \ngentilezas"
print(araucana_linea) 


No 
las 
damas, 
amor, 
no 
gentilezas


In [ ]:
## Su respuesta desde acá.